utils/utils.h

```diff
@@ -262,6 +262,14 @@ static inline bool host_is_lp64(void)
 	return get_elf_class() == ELFCLASS64;
 }
 
+static inline char* has_kernel_opt(char *buf)
+{
+    char* res = strstr(buf, "@kernel");
+    if (!res)
+        res =  strstr(buf, "@k");
```

Comment:
You also need to check whether it's a whole word.  For example it should not allow something like "foo@kernel-program" or "bar@kbs".  I know the original code had the same problem, please fix it.